src/containers/profile/timeline.svelte

```diff
@@ -1,49 +1,80 @@
 <script>
-  import { createEventDispatcher } from 'svelte';
-  import Button from 'ui/button.svelte';
+  import { createEventDispatcher, getContext } from 'svelte';
+  import { Button, Loading } from 'attractions';
+  import { snackbarContextKey } from 'attractions/snackbar';
+  import EmptyState from '@/components/common/empty-state.svelte';
   import TimelineEntry from '@/components/profile/timeline-entry.svelte';
+  import LeaveFeedbackModal from '@/components/projects/view/leave-feedback-modal.svelte';
+  import * as api from '@/utils/api.js';
 
   export let timelinePromises;
-
-  const dispatch = createEventDispatcher();
+  export let competences;
 
   function requestMore() {
     dispatch('more-timeline');
   }
+
+  const leaveFeedbackModal = {
+    open: false,
+    activity: null,
+    application: null,
+    payload: null,
+    async show({ detail: timelinePayload }) {
+      try {
+        leaveFeedbackModal.payload = timelinePayload;
+        const project = await api.json(api.get(`/projects/${timelinePayload.project_id}`));
+        leaveFeedbackModal.activity = project.activities.find(
+          act => act.id === timelinePayload.activity_id,
+        );
+        leaveFeedbackModal.application = leaveFeedbackModal.activity.applications.find(
+          apl => apl.id === timelinePayload.application_id,
+        );
+        leaveFeedbackModal.open = true;
+      } catch (e) {
+        showSnackbar({ props: { text: 'Couldn\'t get information about the project' } });
+        console.error(e);
+      }
+    },
+  };
+
+  const showSnackbar = getContext(snackbarContextKey);
+  const dispatch = createEventDispatcher();
 </script>
 
 <div class="timeline">
-  {#if timelinePromises.length}
+  {#if timelinePromises}
```

Comment:
An empty array is truthy in Javascript. Are you sure you want to remove the `.length`?

Reply:
Am not